### DISCUSSION TITLE
dbus-next backend: Connection was not successful!

### DISCUSSION THREAD

**Discussion by lhammond** [hbldh/bleak#441] [Q&A]
* bleak version: dbus-next-2 branch bleak-0.11.0a1-py3.6.egg
* Python version: 3.6.9
* Operating System: Ubuntu 18.04
* BlueZ version 5.48

### Description

All examples fail with 

`bleak/backends/bluezdbus/client.py", line 321, in connect
    f"Connection was not successful! ({self._device_path})"
bleak.exc.BleakError: Connection was not successful! (/org/bluez/hci0/dev_68_27_19_21_E1_FF)`

```
Traceback (most recent call last):
  File "examples/service_explorer.py", line 67, in <module>
    loop.run_until_complete(run(address, True))
  File "/usr/lib/python3.6/asyncio/base_events.py", line 484, in run_until_complete
    return future.result()
  File "examples/service_explorer.py", line 28, in run
    async with BleakClient(address) as client:
  File "/usr/lib/python3.6/asyncio/coroutines.py", line 110, in __next__
    return self.gen.send(None)
  File "/home/clh/projects/bluedog-edge/pygatt-venv/lib/python3.6/site-packages/bleak-0.11.0a1-py3.6.egg/bleak/backends/client.py", line 59, in __aenter__
    await self.connect()
  File "/usr/lib/python3.6/asyncio/coroutines.py", line 110, in __next__
    return self.gen.send(None)
  File "/home/clh/projects/bluedog-edge/pygatt-venv/lib/python3.6/site-packages/bleak-0.11.0a1-py3.6.egg/bleak/backends/bluezdbus/client.py", line 321, in connect
    f"Connection was not successful! ({self._device_path})"
bleak.exc.BleakError: Connection was not successful! (/org/bluez/hci0/dev_68_27_19_21_E1_FF)
```

I do see a connect/disconnect twice in bluetoothctl output

![image](https://user-images.githubusercontent.com/229602/107054131-bb031580-679d-11eb-8127-d0bb14c9884d.png)

**Comment by dlech** [comment]
What sort of device are you connecting to? 

We had some report a similar issue with an iPhone and it turned out the be that BlueZ was trying to also connect to the battery service in the background but it failed because it required a secure connection or something like that. The workaround was to disable the battery plugin in BlueZ.

**Reply by lhammond** [reply]
@dlech It's a TempTest Blue food thermometer by ThermoWorks

I've been successful in connecting/interacting with all of the device's features with the following
- pygatt
- gattool interactive
- nRF Connect on iPhone and Macbook

I'm currently targeting linux but an ask has been made to support other platforms hence my interest in Bleak.

thank you and let me know if you'd like me to do anything!

**Reply by lhammond** [reply]
Hi David, just following up.
Did you have any further recommendations?
cheers, C

**Reply by dlech** [reply]
You could try logging the Bluetooth packets to see what is really going on. https://bleak.readthedocs.io/en/latest/troubleshooting.html#capture-bluetooth-traffic

**Reply by lhammond** [reply]
this is from examples/service_explorer.py

lemme know if you have any ideas or recommendations .. thanks for your feedback
 
![image](https://user-images.githubusercontent.com/229602/107714039-84943180-6c9a-11eb-8368-b2a84a67c176.png)

![image](https://user-images.githubusercontent.com/229602/107715080-7e9f5000-6c9c-11eb-921b-79957b3c23a7.png)

**Reply by lhammond** [reply]
![image](https://user-images.githubusercontent.com/229602/107715395-2fa5ea80-6c9d-11eb-8d2e-9aad61efe676.png)

**Reply by lhammond** [reply]
rando intel .. but

this search string yields a single result from which the above screenshot comes -> ble "Error: PIN or Key Missing" bleak

https://www.raspberrypi.org/forums/viewtopic.php?t=277035

my kernel is 5.4.0-65-generic

**Comment by lhammond** [comment]
@dlech I moved on with pygatt but happy to further test if you think of something
